dwave_networkx/generators/chimera.py

```diff
@@ -356,147 +357,119 @@ def __init__(self, m, n=None, t=4):
-        Converts a sequence of chimera_index node labels into
-        linear_index node labels, preserving order
-
-        Parameters
-        ----------
-        qlist : sequence of ints
-            The chimera_index node labels
-
-        Returns
-        -------
-        rlist : iterable of tuples
-            The linear_lindex node lables corresponding to qlist
-        """
-
         m, n, t = self.args
-        return (((n*i + j)*2 + u)*t + k for (i, j, u, k) in qlist)
+        for (i, j, u, k) in qlist:
+            yield ((n*i + j)*2 + u)*t + k
 
     def tuples(self, rlist):
+        """Deprecated alias for `iter_linear_to_chimera`."""
+        msg = ('chimera_coordinates.tuples is deprecated and will be removed in '
+               'dwave-networkx 0.9.0, please use '
+               'chimera_coordinates.iter_linear_to_chimera instead')
+        warnings.warn(msg, DeprecationWarning)
+        return self.iter_linear_to_chimera(rlist)
+
+    def iter_linear_to_chimera(self, rlist):
+        """Return an iterator converting linear indices to 4-term Chimera
+        coordinates.
         """
-        Converts a sequence of linear_index node labels into
-        chimera_index node labels, preserving order
-
-        Parameters
-        ----------
-        rlist : sequence of tuples
-            The linear_index node labels
-
-        Returns
-        -------
-        qlist : iterable of ints
-            The chimera_lindex node lables corresponding to rlist
-        """
-
         m, n, t = self.args
         for r in rlist:
             r, k = divmod(r, t)
             r, u = divmod(r, 2)
             i, j = divmod(r, n)
             yield i, j, u, k
 
-    def __pair_repack(self, f, plist):
-        """
-        Flattens a sequence of pairs to pass through `f`, and then
+    @staticmethod
+    def _pair_repack(f, plist):
+        """Flattens a sequence of pairs to pass through `f`, and then
         re-pairs the result.
-
-        Parameters
-        ----------
-        f : callable
-            A function that accepts a sequence and returns a sequence
-        plist:
-            A sequence of pairs
-
-        Returns
-        -------
-        qlist : sequence
-            Equivalent to (tuple(f(p)) for p in plist)
         """
         ulist = f(u for p in plist for u in p)
         for u in ulist:
             v = next(ulist)
             yield u, v
 
     def int_pairs(self, plist):
+        """Deprecated alias for `iter_chimera_to_linear_pairs`."""
+        msg = ('chimera_coordinates.int_pairs is deprecated and will be removed in '
+               'dwave-networkx 0.9.0, please use '
+               'chimera_coordinates.iter_chimera_to_linear_pairs instead')
+        warnings.warn(msg, DeprecationWarning)
+        return self.iter_chimera_to_linear_pairs(plist)
+
+    def iter_chimera_to_linear_pairs(self, plist):
+        """Return an iterator converting pairs of 4-term Chimera coordinates to
+        pairs of linear indices.
```

Comment:
At this point I worry users might find getting the format correct hard without it being specified because it is not intuitive that plist is a list that contains one or more tuples in which each member is a 4-term tuple. If we want to keep it short maybe either a 2nd sentence describing the format or a one-line example?

Reply:
I added an example showing the use